checker/src/main/java/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java

```diff
@@ -371,6 +371,11 @@ public NullnessAnnotatedTypeFactory(BaseTypeChecker checker) {
         NULLABLE_ALIASES.forEach(annotation -> addAliasedTypeAnnotation(annotation, NULLABLE));
         POLYNULL_ALIASES.forEach(annotation -> addAliasedTypeAnnotation(annotation, POLYNULL));
 
+        // Add compatible annotations from alias file
+        addAliasedTypeAnnotation(NonNull.class);
```

Comment:
Every type system needs to explicitly opt-in to using these alias files by having one call of `addAliasedTypeAnnotation` per qualifier, for each qualifier that should support aliases.
That seems rather error-prone to use, which is why you only did this for a few type systems.

Would a more complex file format solve this issue without too much extra effort?
Something like:

````
FQN.canonical.Qualifier = FQN.alias1.Qual1, FQN.alias2.Qual2
FQN.canonical.OtherQualifier = FQN.alias1.OtherQual1, FQN.alias2.OtherQual2
````

This shouldn't be much harder to parse and allows one file for all the different qualifiers. All type systems would support this automatically, without having to specifically opt-in to this.
What do you think?

Reply:
Thinking more about this, why does this need to be an external file at all? Wouldn't it be easier to have a normal command-line flag, e.g. for the above `-AaliasedTypeAnnos=FQN.canonical.Qualifier:FQN.alias1.Qual1,FQN.canonical.Qualifier:FQN.alias2.Qual2,FQN.canonical.OtherQualifier:FQN.alias1.OtherQual1,FQN.canonical.OtherQualifier :FQN.alias2.OtherQual2` or something, with whatever good separators are.

Then, if this makes the command too long, one can simply use the standard `@filename` option to include the options from the given file.

Reply:
Sounds good I like the design. How does the `@filname` work? I want to be able to auto-generate these alias file and even pass it to other projects.